src/mono/wasm/runtime/library_mono.js

```diff
@@ -362,43 +362,97 @@ var MonoSupportLib = {
 			var i = 0;
 			while (i < var_list.length) {
 				let o = var_list [i];
-				const name = o.name;
-				if (name == null || name == undefined) {
+				const this_has_name = o.name !== undefined;
+				let next_has_value_or_get_set = false;
+
+				if (i + 1 < var_list.length) {
+					const next = var_list [i+1];
+					next_has_value_or_get_set = next.value !== undefined || next.get !== undefined || next.set !== undefined;
+				}
+
+				if (!this_has_name || !next_has_value_or_get_set) {
 					i ++;
 					out_list.push (o);
 					continue;
 				}
 
-				if (i + 1 < var_list.length) {
-					o = Object.assign (o, var_list [i + 1]);
-				}
-
+				o = Object.assign (o, var_list [i + 1]);
 				out_list.push (o);
 				i += 2;
 			}
 
 			return out_list;
 		},
 
-		_filter_automatic_properties: function (props) {
-			let names_found = {};
-			let final_var_list = [];
+		_filter_automatic_properties: function (props, accessors_only=false) {
+			// Note: members in @props, have derived class members, followed by
+			//       those from parent classes
 
-			for (var i in props) {
-				var p = props [i];
-				if (p.name in names_found)
-					continue;
+			// Note: Auto-properties have backing fields, named with a special suffix.
+			//       @props here will have the backing field, *and* the getter.
+			//
+			//       But we want to return only one name/value pair:
+			//          [name of the auto-property] = value of the backing field
 
-				if (p.name.endsWith ("k__BackingField"))
-					p.name = p.name.replace ("k__BackingField", "")
-							.replace ('<', '')
-							.replace ('>', '');
+			let getters = {};
+			let all_fields_except_backing_fields = {};
+			let backing_fields = {};
 
-				names_found [p.name] = p.name;
-				final_var_list.push (p);
-			}
+			// Split props into the 3 groups - backing_fields, getters, and all_fields_except_backing_fields
+			props.forEach(p => {
+				if (p.name === undefined) {
+					console.debug(`Bug: Found a member with no name. Skipping it. p: ${JSON.stringify(p)}`);
+					return;
```

Comment:
Should we return or we can just continue the forEach for the next member?

Reply:
It's a lambda, so return here will effectively mean - continue.